### PR TITLE
providing make goal for upheno2-merged.owl

### LINKED ISSUE #8: Provide make goal for upheno2-merged.owl
This is the replacement ticket for https://github.com/obophenotype/upheno-dev/issues/38

We need to iterate over this goal, as it is, as of yet, not clear how to fix this. In the old uPheno, 

MP:123 = HP:123.

In uPheno2, MP:123 sub UPHENO:111, HP:123 sub UPHENO:111, so the equivalence axiom is replaced by a common parent. This drastically changes the way (graph-based) semantic similarity algorithms behave.

This is the key ticket: https://github.com/INCATools/ontology-access-kit/issues/634

@souzadevinicius This has a high priority, but right now, I don't know how to advice you on fixing it.

Can you make sure this does not fall under the radar, and mention it to me every time we meet? (add to your board as high priority)

### PR DESCRIPTION
**Pull request by souzadevinicius**
fixes #8

### CODE REVIEW

File: custom.Makefile [hunk 1]
```diff
@@ -1,2 +1,55 @@
+SPARQLDIR				:=	data/sparql
+
 $(ONTOLOGYDIR)/upheno1-equivalent.owl: $(MIRRORDIR)/upheno1-equivalent.owl
-	robot merge -i $< -o $@
\ No newline at end of file
+	robot merge -i $< -o $@
+
+$(ONTOLOGYDIR)/upheno2-equivalent.owl: $(MIRRORDIR)/upheno2-lattice.owl $(ONTOLOGYDIR)/upheno-mappings-equivalent-class.owl
+	robot query -i $(ONTOLOGYDIR)/upheno-mappings-equivalent-class.owl --update $(SPARQLDIR)/prepare-upheno-mapping.ru \
```
Comment (matentzn) [inline custom.Makefile:7]: Where is prepare-upheno-mapping.ru?

File: custom.Makefile [hunk 2]
```diff
@@ -1,2 +1,55 @@
...
+	merge -i $< -o $@
+
+
+$(TMP_DATA)/upheno_custom_mapping.sssom.tsv: #is created using phenio-toolkit
+	phenio-toolkit lexical-mapping \
+	--species-lexical $(TMP_DATA)/upheno_species_lexical.csv \
```
Comment (matentzn) [inline custom.Makefile:13]: Where is the goal for $(TMP_DATA)/upheno_species_lexical.csv?

File: custom.Makefile [hunk 3]
```diff
@@ -1,2 +1,55 @@
...
+# robot query -i $@
+
+
+$(ONTOLOGYDIR)/phenio-without-abstract.owl: $(MIRRORDIR)/phenio.owl
+	robot merge -i $< \
+	remove --select "BFO:*" --select classes \
+	remove --select "PATO:*" --select classes \
+	-o $@
+
+$(ONTOLOGYDIR)/phenio-without-abstract.db: $(ONTOLOGYDIR)/phenio-without-abstract.owl
```
Comment (matentzn) [inline custom.Makefile:35]: Maybe replace by a more generic goal like https://github.com/monarch-initiative/mondo-ingest/blob/main/src/ontology/mondo-ingest.Makefile#L454?

That way you don't need to repeat it for all ontologies.

File: custom.Makefile [hunk 4]
```diff
@@ -1,2 +1,55 @@
...
+	./odk.sh semsql make $@
+
+$(ONTOLOGYDIR)/upheno2-lattice-without-abstract.owl: $(MIRRORDIR)/upheno2-lattice.owl
+	robot merge -i $< \
+	remove --select "BFO:*" --select classes \
+	remove --select "PATO:*" --select classes \
+	-o $@
+
+$(ONTOLOGYDIR)/upheno2-lattice-without-abstract.db: $(ONTOLOGYDIR)/upheno2-lattice-without-abstract.owl
```
Comment (matentzn) [inline custom.Makefile:44]: See above.

File: custom.Makefile (hunk 2 above)
Comment (matentzn) [reply custom.Makefile:13]: Always add _all dependencies_ for all goals into the make goals! This serves as much as documentation as it helps with debugging later.

**Review by matentzn — CHANGES_REQUESTED** [review]

File: custom.Makefile (hunk 2 above)
Comment (souzadevinicius) [reply custom.Makefile:13]: We need to discuss how upheno_species_lexical.csv is generated

**Review by matentzn — COMMENTED** [review]
Small suggestion, then ready to merge

**Pull request merged**